cmd/kpod/history.go

```diff
@@ -0,0 +1,349 @@
+
+// outputString outputs the information in historyOutputParams
+func outputString(noTrunc, human bool, params historyOutputParams) {
+	var (
+		createdTime string
+		outputSize  string
+	)
+
+	if human {
+		createdTime = outputHumanTime(params.Created) + " ago"
+		outputSize = units.HumanSize(float64(params.Size))
+	} else {
+		createdTime = outputTime(params.Created)
+		outputSize = strconv.FormatInt(params.Size, 10)
+	}
+
+	if !noTrunc {
+		fmt.Printf("%-12.12s\t\t%-16s\t\t%-42.42s\t\t", params.ID, createdTime, params.CreatedBy)
+		fmt.Printf("%-16s\t\t%s\n", outputSize, params.Comment)
+	} else {
+		fmt.Printf("%-64s\t%-18s\t%-60s\t", params.ID, createdTime, params.CreatedBy)
+		fmt.Printf("%-16s\t%s\n\n", outputSize, params.Comment)
+	}
+}
+
+// outputWithTemplate is called when --format is given a template
+func outputWithTemplate(format string, params historyOutputParams, human bool) error {
+	templ, err := template.New("history").Parse(format)
+	if err != nil {
+		return errors.Wrapf(err, "Template parsing error")
+	}
+
+	createdTime := outputTime(params.Created)
+	outputSize := strconv.FormatInt(params.Size, 10)
+
+	if human {
+		createdTime = outputHumanTime(params.Created) + " ago"
+		outputSize = units.HumanSize(float64(params.Size))
+	}
+
+	// templParams is used to store the info from params and the time and
+	// size that have been converted to type string for when the human flag
+	// is set
+	templParams := struct {
+		ID        string
+		Created   string
+		CreatedBy string
+		Size      string
+		Comment   string
+	}{
+		params.ID,
+		createdTime,
+		params.CreatedBy,
+		outputSize,
+		params.Comment,
+	}
+
+	if err = templ.Execute(os.Stdout, templParams); err != nil {
+		return err
+	}
+	fmt.Println()
+	return nil
+}
+
+// outputTime displays the time stamp in "2017-06-20T20:24:10Z" format
+func outputTime(tm *time.Time) string {
+	return tm.Format(time.RFC3339)
+}
+
+// outputHumanTime displays the time elapsed since creation
+func outputHumanTime(tm *time.Time) string {
+	return units.HumanDuration(time.Since(*tm))
+}
+
+// createJSON retrieves the history of the image and returns a JSON object
+func createJSON(store storage.Store, opts historyOptions) ([]byte, error) {
+	var (
+		size     int64
+		img      *storage.Image
+		imageID  string
+		layerAll []historyOutputParams
+	)
+
+	ref, err := is.Transport.ParseStoreReference(store, opts.image)
+	if err != nil {
+		return nil, errors.Errorf("error parsing reference to image %q: %v", opts.image, err)
+	}
+
+	img, err = is.Transport.GetStoreImage(store, ref)
+	if err != nil {
+		return nil, errors.Errorf("no such image %q: %v", opts.image, err)
+	}
+
+	systemContext := getSystemContext("")
+
+	src, err := ref.NewImage(systemContext)
+	if err != nil {
+		return nil, errors.Errorf("Error instantiating image %q: %v", opts.image, err)
+	}
+
+	oci, err := src.OCIConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	history := oci.History
+	layers := src.LayerInfos()
+	count := 1
+
```

Comment:
nit: count should be co-located with the following blog so you could remove the empty line.